### DISCUSSION TITLE
How do you turn off formatting for certain file extensions?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [codesandbox/codesandbox-client#5526] [Q&A] [answered]
It seems the editor has some incorrect formatting assumptions for *.hbs files.
I'm working on a custom template, is there a way to, in the csb config to disable formatting for certain file extensions?

**Comment by smhutch** [comment, marked as answer]
We don't support disabling formatting specifically for handlebars files, but there are two approaches you may want to take.

## Approach 1: disable auto-formatting

1. Open the command palette, search for "User settings"
2. In the settings, search for `editor.format`
3. Uncheck "Editor: Format on Save"

## Approach 2: disable for HTML files (which also catches handlebars)

1. Open the command palette, search for "User settings"
2. In the settings, search for `html.format`
3. Uncheck "Enable/disable default HTML formatter."

Alternatively, you may also add a `.vscode/settings.json` file to your sandbox:

```json
{
  "html.format.enable": false
}
```

**Reply by NullVoxPopuli** [reply]
can this be done in the sandbox settings json file in a template?
I'm working on a template and need control of the settings folks have enabled when they make sandboxes from my template

**Reply by NullVoxPopuli** [reply]
or, would that be `.vscode/settings.json`?

**Reply by smhutch** [reply]
I believe adding it to `.vscode/settings.json` in your template will also have the same result for sandboxes created from your template.
